docs/source/image_process.mdx

```diff
@@ -0,0 +1,133 @@
+
+```
+folder/dog/golden_retriever.png
+folder/dog/german_shepherd.png
+folder/dog/chihuahua.png
+
+folder/cat/maine_coon.png
+folder/cat/bengal.png
+folder/cat/birman.png
+```
+
+Then you can load your dataset by specifying `imagefolder` and the directory of your dataset in `data_dir`:
+
+```py
+>>> from datasets import load_dataset
+>>> dataset = load_dataset("imagefolder", data_dir="/path/to/folder")
+```
+
+Load remote datasets from their URLs with the `data_files` parameter:
+
+```py
+>>> dataset = load_dataset("imagefolder", data_files="https://download.microsoft.com/download/3/E/1/3E1C3F21-ECDB-4869-8368-6DEBA77B919F/kagglecatsanddogs_3367a.zip", split="train")
+```
+
+## Data augmentation
+
+Adding data augmentations to a dataset is common to prevent overfitting and achieve better performance. You can use any library or package you want to apply the augmentations. This guide will use the transforms from [torchvision](https://pytorch.org/vision/stable/transforms.html).
+
+Add the [`ColorJitter`](https://pytorch.org/vision/stable/transforms.html#torchvision.transforms.ColorJitter) transform to change the color properties of the image randomly:
+
+```py
+>>> from torchvision.transforms import Compose, ColorJitter, ToTensor
+
+>>> jitter = Compose(
+... [
+...  ColorJitter(brightness=0.25, contrast=0.25, saturation=0.25, hue=0.7),
+...  ToTensor(),
+...  ]
+... )
+```
+
+Create a function to apply the `ColorJitter` transform to an image:
+
+```py
+>>> def transforms(examples):
+...     examples["pixel_values"] = [jitter(image.convert("RGB")) for image in examples["image"]]
+...     return examples
+```
+
+Then you can use the [`~Dataset.set_transform`] function to apply the transformation on-the-fly to consume less disk space. You should use [`~Dataset.set_transform`] if you only need to access the examples once:
+
+```py
+>>> dataset.set_transform(transforms)
+```
+
+Now visualize the results of the `ColorJitter` transform:
+
+```py
+>>> import numpy as np
+>>> import matplotlib.pyplot as plt
+
+>>> img = dataset[0]["pixel_values"]
+>>> plt.imshow(img.permute(1, 2, 0))
+```
+
+![image_process_jitter](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/image_process_jitter.png)
+
+## Map
+
+[`~Dataset.map`] can also apply transforms over an entire dataset. When you use [`~Dataset.map`] to apply transforms, it generates a cache file. This saves time because you don't have to execute the same transform twice. You should use [`~Dataset.map`] for operations you only run once per training - like resizing an image - instead of using it for operations that are executed for each epoch, like data augmentations.
+
+To apply the same `ColorJitter` transform from above with [`~Dataset.map`], you need to set `batched=True`:
+
+```py
+>>> processed_food = dataset.map(transforms, batched=True)
+```
```

Comment:
One last nit:

Let's also add a transform that returns `PIL.Image` objects to show it's possible to cache them as well (e.g. resize).

Additionally, it makes sense to add a tip explaining how to reduce memory requirements of `map` with `batch_size` (controls the number of examples that are processed in a single call to the transform function) and `writer_batch_size` (controls the number of the transformed examples that are kept in memory before they are "flushed" into an arrow cache file) because `batch_size`/`writer_batch_size` is 1000 by default so it can be expensive to store 1000 images in RAM.